datalake/src/main/java/com/sequenceiq/datalake/cm/RangerCloudIdentityService.java

```diff
@@ -24,21 +27,87 @@
+    private RangerCloudIdentitySyncStatus sucessfulCloudIdSyncStatus(String message) {
+        RangerCloudIdentitySyncStatus status = new RangerCloudIdentitySyncStatus();
+        status.setState(RangerCloudIdentitySyncState.SUCCESS);
+        status.setStatusReason(message);
+        return status;
+    }
+
+    private RangerCloudIdentitySyncStatus failedRangerCloudIdentitySyncStatus(String message) {
+        RangerCloudIdentitySyncStatus status = new RangerCloudIdentitySyncStatus();
+        status.setState(RangerCloudIdentitySyncState.FAILED);
+        status.setStatusReason(message);
+        return status;
+    }
+
+    private RangerCloudIdentitySyncStatus setAzureCloudIdentityMapping(String envCrn, SdxCluster sdxCluster, Map<String, String> azureUserMapping) {
         String stackCrn = sdxCluster.getStackCrn();
-        LOGGER.info("Updating azure cloud id mappings for datalake stack crn = {}", stackCrn);
+        LOGGER.info("Updating azure cloud id mappings for envCrn = {}, datalake stack crn = {}", envCrn, stackCrn);
         try {
-            clouderaManagerRangerUtil.setAzureCloudIdentityMapping(stackCrn, azureUserMapping);
+            if (!clouderaManagerRangerUtil.isCloudIdMappingSupported(sdxCluster.getStackCrn())) {
+                return sucessfulCloudIdSyncStatus("The datalake does not support cloud identity sync. Sync request is ignored.");
+            }
+            Optional<ApiCommand> apiCommand = clouderaManagerRangerUtil.setAzureCloudIdentityMapping(stackCrn, azureUserMapping);
+            if (apiCommand.isEmpty()) {
+                return sucessfulCloudIdSyncStatus("Sucessfully synced, no role refresh required");
+            }
+            return toRangerCloudIdentitySyncStatus(apiCommand.get());
         } catch (ApiException e) {
             LOGGER.error("Encountered api exception", e);
-            throw new RangerCloudIdentitySyncException("Encountered api exception", e);
+            return failedRangerCloudIdentitySyncStatus("Encountered cloudera manager api exception");
+        }
+    }
+
+    private Optional<SdxCluster> getSdxCluster(String envCrn) {
+        List<SdxCluster> sdxClusters = sdxService.listSdxByEnvCrn(envCrn);
+        if (sdxClusters.isEmpty()) {
+            return Optional.empty();
+        } else if (sdxClusters.size() > 1) {
+            LOGGER.error("Multiple datalakes per environment not supported, environmentCrn = {}", envCrn);
+            throw new IllegalStateException("Multiple datalakes per environment not supported");
+        }
+        return Optional.of(Iterables.getOnlyElement(sdxClusters));
+    }
+
+    public RangerCloudIdentitySyncStatus setAzureCloudIdentityMapping(String envCrn, Map<String, String> azureUserMapping) {
+        Optional<SdxCluster> sdxCluster = getSdxCluster(envCrn);
+        if (sdxCluster.isEmpty()) {
+            return sucessfulCloudIdSyncStatus("No datlakes associated with the environment.");
```

Comment:
typo: datalakes
also, this doesn't sound like a successful scenario. Is there a better status to describe what happened?